src/olympia/scanners/tasks.py

```diff
@@ -229,17 +229,24 @@ def _run_yara_for_path(scanner_result, path, definition=None):
                     scanner=YARA, is_active=True, definition__isnull=False
                 ).values_list('definition', flat=True)
             )
-
-        rules = yara.compile(source=definition)
+        # Initialize external variables so that compilation works, we'll
+        # override them later when matching.
+        externals = {
+            'filename': '',
```

Comment:
I don't think this is necessarily a good idea, especially given that the real use case is different than what's described in the issue. The use case seems to be about detecting `manifest.json` and locale files.

While the former can be retrieved easily, it is definitely tricky for locale files but we could do something a bit different. We could inject three variables instead of the filename (because we should really write rules that apply to the content of a file and not based on its name, which can be anything).

How about we inject the following instead?

```
'is_json_file': zip_info.filename.endswith('.json'),
'is_manifest': zip_info.filename == 'manifest.json',  # I guess there is a better way to compare strings in python
'is_locale_file': zip_info.filename.contains('_locales') and zip_info.filename.endswith('messages.json'),  # there is probably a better way to write that, see: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Internationalization#Localized_string_selection
```

Reply:
https://github.com/mozilla/addons-server/pull/13987/commits/6c5391f3a3a922b0a0163ae9e641cba4f4aa2451 does that. I am not a super fan of how rigid this is - that's why I originally went for just `filename` - but we can try this out and see if that covers our needs.